pkg/reconciler/reconciler.go

```diff
@@ -745,7 +765,7 @@ func (r *Reconciler) setupWatches(mgr ctrl.Manager, c controller.Controller) err
 	); err != nil {
 		return err
 	}
-
+	// why we are adding the secret here?
 	secret := &corev1.Secret{}
 	secret.SetGroupVersionKind(schema.GroupVersionKind{
 		Group:   "",
```

Comment:
why we are adding the secret here?

Reply:
Because Helm will create a release secret. So we want to watch that and reconcile if it is deleted or changed. Sometimes that can cause an unrecoverable error, but by watching it, we can bubble that problem up to the CR status.